runners/direct-java/src/test/java/org/apache/beam/runners/direct/ViewOverrideFactoryTest.java

```diff
@@ -0,0 +1,122 @@
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.View.CreatePCollectionView;
+import org.apache.beam.sdk.util.PCollectionViews;
+import org.apache.beam.sdk.util.WindowingStrategy;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionView;
+import org.hamcrest.Matchers;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.omg.CORBA.WStringSeqHelper;
+
+/** Tests for {@link ViewOverrideFactory}. */
+@RunWith(JUnit4.class)
+public class ViewOverrideFactoryTest implements Serializable {
+  @Rule
+  public transient TestPipeline p = TestPipeline.create().enableAbandonedNodeEnforcement(false);
+
+  private transient ViewOverrideFactory<Integer, List<Integer>> factory =
+      new ViewOverrideFactory<>();
+
+  @Test
+  public void replacementSucceeds() {
+    PCollection<Integer> ints = p.apply("CreateContents", Create.of(1, 2, 3));
+    final PCollectionView<List<Integer>> view =
+        PCollectionViews.listView(p, WindowingStrategy.globalDefault(), ints.getCoder());
+    PCollectionView<List<Integer>> afterReplacement =
+        ints.apply(
+            factory.getReplacementTransform(
+                CreatePCollectionView.<Integer, List<Integer>>of(view)));
+
+    PCollection<Set<Integer>> outputViewContents =
+        p.apply("CreateSingleton", Create.of(0))
+            .apply(
+                "OutputContents",
+                ParDo.of(
+                        new DoFn<Integer, Set<Integer>>() {
+                          @ProcessElement
+                          public void outputSideInput(ProcessContext context) {
+                            context.output(ImmutableSet.copyOf(context.sideInput(view)));
+                          }
+                        })
+                    .withSideInputs(view));
+    PAssert.thatSingleton(outputViewContents).isEqualTo(ImmutableSet.of(1, 2, 3));
+
+    p.run();
+  }
+
+  @Test
+  public void replacementGetViewReturnsOriginal() {
+    PCollection<Integer> ints = p.apply("CreateContents", Create.of(1, 2, 3));
+    final PCollectionView<List<Integer>> view =
+        PCollectionViews.listView(p, WindowingStrategy.globalDefault(), ints.getCoder());
+    PTransform<PCollection<Integer>, PCollectionView<List<Integer>>> replacement =
+        factory.getReplacementTransform(CreatePCollectionView.<Integer, List<Integer>>of(view));
+    ints.apply(replacement);
+    final AtomicBoolean writeViewVisited = new AtomicBoolean();
+    p.traverseTopologically(
+        new PipelineVisitor.Defaults() {
+          @Override
+          public void visitPrimitiveTransform(Node node) {
+            assertThat(
+                "WriteView should be the last primtitive in the graph",
```

Comment:
* typo: primitive
* why?
* this condition is actually testing that there's exactly one WriteView transform, not that it's "last"